test/src/unittests/rhythm/test_loopbpmconfidence.py

```diff
@@ -0,0 +1,212 @@
+        confidence = LoopBpmConfidence()(audio[len01:len90], bpmEstimate)                         
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+
+        bpmEstimate = 120
+        expectedConfidence =  0.36 # rounded from measured 0.3631746172904968
+        confidence = LoopBpmConfidence()(audio[len01:len90], bpmEstimate)           
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+        
+        bpmEstimate = 70 
+        expectedConfidence =  0.79 # rounded from measured 0.7951852083206177
+        confidence = LoopBpmConfidence()(audio[len01:len90], bpmEstimate)
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+
+
+    """
+    The confidence returned is based on comparing the duration of the audio signal with 
+    multiples of the BPM estimate. 
+    The algorithm considers edge cases in which the signal includes non-musical 
+    silence at the start or end of the audio signal (or both) and returns the confidence for the best fit.
+    For more details refer to paper: https://repositori.upf.edu/handle/10230/33113
+
+    Additional Tests were made with following loop
+    "458538__flowerpunkchip__mellifluousanonymous-120bpm-loop-youareevil.mp3'"
+    https://freesound.org/people/Flowerpunkchip/sounds/458538/
+
+    Front end slience Confidence is  0.9915646314620972
+    Back end silence Confidence  is  0.9915646314620972
+    Silence Both end confidence is  0.9915646314620972    
+    """
+    def testSilentEdges(self):
+
+        audio = MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'techno_loop.wav'))() 
+        bpmEstimate = 125
+        confidence = LoopBpmConfidence()(audio, bpmEstimate)              
+        lenSilence = int(0.01*len(audio)) # Choose a silent length 1% of audio
+        silentAudio = zeros(lenSilence)
+
+        # case 1: there is non-musical* silence before the loop starts
+        signal1  = numpy.append(silentAudio, audio)
+        confidence = LoopBpmConfidence()(signal1, bpmEstimate)              
+        self.assertGreater(confidence, 0.9) 
+
+        # Silent Edges dont always work.
+        # When adding silence to the end of this loop the confidence drops.
+        # This isnt a big problem since it is just a way 
+        # to try to better estimate confidence if the loop is not well cut.
+        # In any case, the "mellifluousanonymous" loop mentioned above
+        # works well with silence added at the end.
+
+        # case 2: there is non-musical silence after the loop ends        
+        signal2  = numpy.append(audio,silentAudio)
+        confidence = LoopBpmConfidence()(signal2, bpmEstimate)  
+        self.assertGreater(confidence, 0.27) 
+
+
+        # case 3: there is non-musical silence at both ends
+        signal3 = numpy.append(signal1, silentAudio)
+        confidence = LoopBpmConfidence()(signal3, bpmEstimate)              
+        self.assertGreater(confidence, 0.27) 
```

Comment:
As we have commented before, there are situations when adding a silence segment does not change the estimated confidence. This isn't covered by the test.

Reply:
with the updates made to the groud truth , the addition of silence in the scenarios (beginning, end , both) now yields higher confidence.